### ISSUE TITLE
Quantization with `return_inverse` and labels outputs wrong inverse maps

### ISSUE THREAD

**Issue by kumuji** [NVIDIA/MinkowskiEngine#211]
I found strange behavior, when I try to pass coordinates, features and labels, and get inverse maps I get wrong inverse maps.
You can find the result in code snippet below.
(version 0.4.3)
```python
import MinkowskiEngine as ME
import numpy as np


coords = np.array(
    [
        [0, 0, 0],
        [0, 0, 1],
        [0, 0, 2],
        [0, 0, 3],
        [0, 0, 4],
        [0, 0, 5],
        [0, 0, 6],
        [0, 0, 7],
        [0, 0, 8],
        [0, 0, 9],
    ],
    dtype=np.float32,
)
feats = np.empty((10, 3), dtype=np.float32)
voxel_size = 2
labels = np.array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9], dtype=np.int32)

voxelization_dict = {
    "coords": coords,
    "feats": feats,
    "labels": labels,
    "ignore_label": -100,
    "return_index": True,
    "return_inverse": True,
    "quantization_size": voxel_size,
}

unique_map, inverse_map = ME.utils.sparse_quantize(**voxelization_dict)
# unique_map = [8, 4, 2, 0, 6]
# BUG? inverse_map = [-100, -100, -100, -100, -100]
# as in documentation: https://nvidia.github.io/MinkowskiEngine/utils.html#MinkowskiEngine.utils.sparse_quantize
unique_coords = coords[unique_map]
assert unique_coords[inverse_map] == coords  # fails
```

**Comment by chrischoy** [comment]
You passed the ` "ignore_label": -100,` --> if there are label collision, it will return the ignore label, which is -100.

**Comment by kumuji** [comment]
But it is returning the label, not the index, that's the problem

**Comment by chrischoy** [comment]
Ah sorry, you are right. Let me fix the error.

**Comment by chrischoy** [comment]
It is fixed on the v0.5

```python
import MinkowskiEngine as ME
import numpy as np


coords = np.array(
    [
        [0, 0, 0],
        [0, 0, 1],
        [0, 0, 2],
        [0, 0, 3],
        [0, 0, 4],
        [0, 0, 5],
        [0, 0, 6],
        [0, 0, 7],
        [0, 0, 8],
        [0, 0, 9],
    ],
    dtype=np.float32,
)
feats = np.empty((10, 3), dtype=np.float32)
quant_size = 2
labels = np.array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9], dtype=np.int32)

input_dict = {
    "coordinates": coords,
    "features": feats,
    "labels": labels,
    "ignore_label": -100,
    "return_index": True,
    "return_inverse": True,
    "quantization_size": quant_size,
}

unique_coords, _, _, unique_map, inverse_map = ME.utils.sparse_quantize(**input_dict)
assert np.allclose(unique_coords[inverse_map], np.floor(coords / quant_size))  # fails
```
